modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BTypes.java

```diff
@@ -145,9 +145,16 @@ public static BType resolveType(SimpleTypeName typeName, SymbolScope symbolScope
         // If bType is not null, then element type of this arrays type is available.
         // We should define the arrays type here.
         if (bType != null) {
+            SimpleTypeName childSimpleType = new SimpleTypeName(typeName.getName(), typeName.getPackagePath(), true);
+            childSimpleType.setPkgPath(typeName.getPackagePath());
+            childSimpleType.setDimensions(typeName.getDimensions() - 1);
+            BTypes.resolveType(childSimpleType, symbolScope, location);
+
             BArrayType bArrayType = new BArrayType(typeName.getSymbolName().getName(),
-                    bType, typeName.getPackagePath(), bType.getSymbolScope());
+                    BTypes.resolveType(childSimpleType, symbolScope, location),
```

Comment:
Sorry, it was added accidentally and removed it from the code. Thank you for pointing out.